{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Installing FastQC and getting it to run via command line"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "            FastQC - A high throughput sequence QC analysis tool\r\n",
      "\r\n",
      "SYNOPSIS\r\n",
      "\r\n",
      "\tfastqc seqfile1 seqfile2 .. seqfileN\r\n",
      "\r\n",
      "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \r\n",
      "           [-c contaminant file] seqfile1 .. seqfileN\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "\r\n",
      "    FastQC reads a set of sequence files and produces from each one a quality\r\n",
      "    control report consisting of a number of different modules, each one of \r\n",
      "    which will help to identify a different potential type of problem in your\r\n",
      "    data.\r\n",
      "    \r\n",
      "    If no files to process are specified on the command line then the program\r\n",
      "    will start as an interactive graphical application.  If files are provided\r\n",
      "    on the command line then the program will run with no user interaction\r\n",
      "    required.  In this mode it is suitable for inclusion into a standardised\r\n",
      "    analysis pipeline.\r\n",
      "    \r\n",
      "    The options for the program as as follows:\r\n",
      "    \r\n",
      "    -h --help       Print this help file and exit\r\n",
      "    \r\n",
      "    -v --version    Print the version of the program and exit\r\n",
      "    \r\n",
      "    -o --outdir     Create all output files in the specified output directory.\r\n",
      "                    Please note that this directory must exist as the program\r\n",
      "                    will not create it.  If this option is not set then the \r\n",
      "                    output file for each sequence file is created in the same\r\n",
      "                    directory as the sequence file which was processed.\r\n",
      "                    \r\n",
      "    --casava        Files come from raw casava output. Files in the same sample\r\n",
      "                    group (differing only by the group number) will be analysed\r\n",
      "                    as a set rather than individually. Sequences with the filter\r\n",
      "                    flag set in the header will be excluded from the analysis.\r\n",
      "                    Files must have the same names given to them by casava\r\n",
      "                    (including being gzipped and ending with .gz) otherwise they\r\n",
      "                    won't be grouped together correctly.\r\n",
      "                    \r\n",
      "    --nano          Files come from nanopore sequences and are in fast5 format. In\r\n",
      "                    this mode you can pass in directories to process and the program\r\n",
      "                    will take in all fast5 files within those directories and produce\r\n",
      "                    a single output file from the sequences found in all files.                    \r\n",
      "                    \r\n",
      "    --nofilter      If running with --casava then don't remove read flagged by\r\n",
      "                    casava as poor quality when performing the QC analysis.\r\n",
      "                   \r\n",
      "    --extract       If set then the zipped output file will be uncompressed in\r\n",
      "                    the same directory after it has been created.  By default\r\n",
      "                    this option will be set if fastqc is run in non-interactive\r\n",
      "                    mode.\r\n",
      "                    \r\n",
      "    -j --java       Provides the full path to the java binary you want to use to\r\n",
      "                    launch fastqc. If not supplied then java is assumed to be in\r\n",
      "                    your path.\r\n",
      "                   \r\n",
      "    --noextract     Do not uncompress the output file after creating it.  You\r\n",
      "                    should set this option if you do not wish to uncompress\r\n",
      "                    the output when running in non-interactive mode.\r\n",
      "                    \r\n",
      "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\r\n",
      "                    show data for every base in the read.  WARNING: Using this\r\n",
      "                    option will cause fastqc to crash and burn if you use it on\r\n",
      "                    really long reads, and your plots may end up a ridiculous size.\r\n",
      "                    You have been warned!\r\n",
      "                    \r\n",
      "    --min_length    Sets an artificial lower limit on the length of the sequence\r\n",
      "                    to be shown in the report.  As long as you set this to a value\r\n",
      "                    greater or equal to your longest read length then this will be\r\n",
      "                    the sequence length used to create your read groups.  This can\r\n",
      "                    be useful for making directly comaparable statistics from \r\n",
      "                    datasets with somewhat variable read lengths.\r\n",
      "                    \r\n",
      "    -f --format     Bypasses the normal sequence file format detection and\r\n",
      "                    forces the program to use the specified format.  Valid\r\n",
      "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\r\n",
      "                    \r\n",
      "    -t --threads    Specifies the number of files which can be processed\r\n",
      "                    simultaneously.  Each thread will be allocated 250MB of\r\n",
      "                    memory so you shouldn't run more threads than your\r\n",
      "                    available memory will cope with, and not more than\r\n",
      "                    6 threads on a 32 bit machine\r\n",
      "                  \r\n",
      "    -c              Specifies a non-default file which contains the list of\r\n",
      "    --contaminants  contaminants to screen overrepresented sequences against.\r\n",
      "                    The file must contain sets of named contaminants in the\r\n",
      "                    form name[tab]sequence.  Lines prefixed with a hash will\r\n",
      "                    be ignored.\r\n",
      "\r\n",
      "    -a              Specifies a non-default file which contains the list of\r\n",
      "    --adapters      adapter sequences which will be explicity searched against\r\n",
      "                    the library. The file must contain sets of named adapters\r\n",
      "                    in the form name[tab]sequence.  Lines prefixed with a hash\r\n",
      "                    will be ignored.\r\n",
      "                    \r\n",
      "    -l              Specifies a non-default file which contains a set of criteria\r\n",
      "    --limits        which will be used to determine the warn/error limits for the\r\n",
      "                    various modules.  This file can also be used to selectively \r\n",
      "                    remove some modules from the output all together.  The format\r\n",
      "                    needs to mirror the default limits.txt file found in the\r\n",
      "                    Configuration folder.\r\n",
      "                    \r\n",
      "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\r\n",
      "                    module. Specified Kmer length must be between 2 and 10. Default\r\n",
      "                    length is 7 if not specified.\r\n",
      "                    \r\n",
      "   -q --quiet       Supress all progress messages on stdout and only report errors.\r\n",
      "   \r\n",
      "   -d --dir         Selects a directory to be used for temporary files written when\r\n",
      "                    generating report images. Defaults to system temp directory if\r\n",
      "                    not specified.\r\n",
      "                    \r\n",
      "BUGS\r\n",
      "\r\n",
      "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\r\n",
      "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\r\n",
      "                   \r\n",
      "    \r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/FastQC/fastqc -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 5% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 10% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 15% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 20% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 25% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 30% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 35% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 40% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 45% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 50% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 55% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 60% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 65% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 70% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 75% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 80% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 85% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 90% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Approx 95% complete for Good-lib1_S1_L001_R1_001.fastq\n",
      "Analysis complete for Good-lib1_S1_L001_R1_001.fastq\n"
     ]
    }
   ],
   "source": [
    "#Visualize the Forward Reads (R1)\n",
    "!/Applications/FastQC/fastqc \\\n",
    "/Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Raw/Good-lib1_S1_L001_R1_001.fastq \\\n",
    "-o /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Raw/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 5% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 10% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 15% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 20% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 25% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 30% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 35% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 40% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 45% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 50% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 55% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 60% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 65% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 70% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 75% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 80% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 85% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 90% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Approx 95% complete for Good-lib1_S1_L001_R2_001.fastq\n",
      "Analysis complete for Good-lib1_S1_L001_R2_001.fastq\n"
     ]
    }
   ],
   "source": [
    "#Visualize the Reverse Reads (R2)\n",
    "!/Applications/FastQC/fastqc \\\n",
    "/Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Raw/Good-lib1_S1_L001_R2_001.fastq \\\n",
    "-o /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Raw/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Raw/Good-lib1_S1_L001_R1_001.fastq\n",
      "Reverse reads file.................: /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Raw/Good-lib1_S1_L001_R2_001.fastq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 350\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 0.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 1\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287156\n",
      "  C: 0.207011\n",
      "  G: 0.216746\n",
      "  T: 0.289087\n",
      "  31536 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,844,518 / 2,386,309 (77.296%)\n",
      "Discarded reads ...................: 541,791 / 2,386,309 (22.704%)\n",
      "Not assembled reads ...............: 0 / 2,386,309 (0.000%)\n",
      "Assembled reads file...............: /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Merged/Lib-1.fastq.assembled.fastq\n",
      "Discarded reads file...............: /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Merged/Lib-1.fastq.discarded.fastq\n",
      "Unassembled forward reads file.....: /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Merged/Lib-1.fastq.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Merged/Lib-1.fastq.unassembled.reverse.fastq\n"
     ]
    }
   ],
   "source": [
    "#pairing the forward and reverse files in pear\n",
    "#added a max length of 350 and added quality control\n",
    "!pear \\\n",
    "-f /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Raw/Good-lib1_S1_L001_R1_001.fastq \\\n",
    "-r /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Raw/Good-lib1_S1_L001_R2_001.fastq \\\n",
    "-n 350 \\\n",
    "-u 0 \\\n",
    "-o /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Merged/Lib-1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The forward and reverse Fastq files are now merged. A max length of 350 and quality control were added. This resulted in 4 fastq files in the \"Merged\" folder of the repository. 1 assembled, 1 discarded, and 2 unassembled. The assembled file will be the one used moving forward."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The next step is to convert FASTQ to FASTA using seqtk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Usage:   seqtk <command> <arguments>\r\n",
      "Version: 1.3-r106\r\n",
      "\r\n",
      "Command: seq       common transformation of FASTA/Q\r\n",
      "         comp      get the nucleotide composition of FASTA/Q\r\n",
      "         sample    subsample sequences\r\n",
      "         subseq    extract subsequences from FASTA/Q\r\n",
      "         fqchk     fastq QC (base/quality summary)\r\n",
      "         mergepe   interleave two PE FASTA/Q files\r\n",
      "         trimfq    trim FASTQ using the Phred algorithm\r\n",
      "\r\n",
      "         hety      regional heterozygosity\r\n",
      "         gc        identify high- or low-GC regions\r\n",
      "         mutfa     point mutate FASTA at specified positions\r\n",
      "         mergefa   merge two FASTA/Q files\r\n",
      "         famask    apply a X-coded FASTA to a source FASTA\r\n",
      "         dropse    drop unpaired from interleaved PE FASTA/Q\r\n",
      "         rename    rename sequence names\r\n",
      "         randbase  choose a random base from hets\r\n",
      "         cutN      cut sequence at long N\r\n",
      "         listhet   extract the position of each het\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!seqtk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "!seqtk \\\n",
    "seq /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Merged/Lib-1.fastq.assembled.fastq > /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/FASTA/Lib-1.assembled.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove tags and split into 16 individual sequences using cutadapt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cutadapt version 1.17\r\n",
      "Copyright (C) 2010-2018 Marcel Martin <marcel.martin@scilifelab.se>\r\n",
      "\r\n",
      "cutadapt removes adapter sequences from high-throughput sequencing reads.\r\n",
      "\r\n",
      "Usage:\r\n",
      "    cutadapt -a ADAPTER [options] [-o output.fastq] input.fastq\r\n",
      "\r\n",
      "For paired-end reads:\r\n",
      "    cutadapt -a ADAPT1 -A ADAPT2 [options] -o out1.fastq -p out2.fastq in1.fastq in2.fastq\r\n",
      "\r\n",
      "Replace \"ADAPTER\" with the actual sequence of your 3' adapter. IUPAC wildcard\r\n",
      "characters are supported. The reverse complement is *not* automatically\r\n",
      "searched. All reads from input.fastq will be written to output.fastq with the\r\n",
      "adapter sequence removed. Adapter matching is error-tolerant. Multiple adapter\r\n",
      "sequences can be given (use further -a options), but only the best-matching\r\n",
      "adapter will be removed.\r\n",
      "\r\n",
      "Input may also be in FASTA format. Compressed input and output is supported and\r\n",
      "auto-detected from the file name (.gz, .xz, .bz2). Use the file name '-' for\r\n",
      "standard input/output. Without the -o option, output is sent to standard output.\r\n",
      "\r\n",
      "Citation:\r\n",
      "\r\n",
      "Marcel Martin. Cutadapt removes adapter sequences from high-throughput\r\n",
      "sequencing reads. EMBnet.Journal, 17(1):10-12, May 2011.\r\n",
      "http://dx.doi.org/10.14806/ej.17.1.200\r\n",
      "\r\n",
      "Use \"cutadapt --help\" to see all command-line options.\r\n",
      "See http://cutadapt.readthedocs.io/ for full documentation.\r\n",
      "\r\n",
      "Options:\r\n",
      "  --version             show program's version number and exit\r\n",
      "  -h, --help            show this help message and exit\r\n",
      "  --debug               Print debugging information.\r\n",
      "  -f FORMAT, --format=FORMAT\r\n",
      "                        Input file format; can be either 'fasta', 'fastq' or\r\n",
      "                        'sra-fastq'. Ignored when reading csfasta/qual files.\r\n",
      "                        Default: auto-detect from file name extension.\r\n",
      "  -j CORES, --cores=CORES\r\n",
      "                        Number of CPU cores to use. Default: 1\r\n",
      "\r\n",
      "  Finding adapters:\r\n",
      "    Parameters -a, -g, -b specify adapters to be removed from each read\r\n",
      "    (or from the first read in a pair if data is paired). If specified\r\n",
      "    multiple times, only the best matching adapter is trimmed (but see the\r\n",
      "    --times option). When the special notation 'file:FILE' is used,\r\n",
      "    adapter sequences are read from the given FASTA file.\r\n",
      "\r\n",
      "    -a ADAPTER, --adapter=ADAPTER\r\n",
      "                        Sequence of an adapter ligated to the 3' end (paired\r\n",
      "                        data: of the first read). The adapter and subsequent\r\n",
      "                        bases are trimmed. If a '$' character is appended\r\n",
      "                        ('anchoring'), the adapter is only found if it is a\r\n",
      "                        suffix of the read.\r\n",
      "    -g ADAPTER, --front=ADAPTER\r\n",
      "                        Sequence of an adapter ligated to the 5' end (paired\r\n",
      "                        data: of the first read). The adapter and any\r\n",
      "                        preceding bases are trimmed. Partial matches at the 5'\r\n",
      "                        end are allowed. If a '^' character is prepended\r\n",
      "                        ('anchoring'), the adapter is only found if it is a\r\n",
      "                        prefix of the read.\r\n",
      "    -b ADAPTER, --anywhere=ADAPTER\r\n",
      "                        Sequence of an adapter that may be ligated to the 5'\r\n",
      "                        or 3' end (paired data: of the first read). Both types\r\n",
      "                        of matches as described under -a und -g are allowed.\r\n",
      "                        If the first base of the read is part of the match,\r\n",
      "                        the behavior is as with -g, otherwise as with -a. This\r\n",
      "                        option is mostly for rescuing failed library\r\n",
      "                        preparations - do not use if you know which end your\r\n",
      "                        adapter was ligated to!\r\n",
      "    -e RATE, --error-rate=RATE\r\n",
      "                        Maximum allowed error rate as value between 0 and 1\r\n",
      "                        (no. of errors divided by length of matching region).\r\n",
      "                        Default: 0.1 (=10%)\r\n",
      "    --no-indels         Allow only mismatches in alignments. Default: allow\r\n",
      "                        both mismatches and indels\r\n",
      "    -n COUNT, --times=COUNT\r\n",
      "                        Remove up to COUNT adapters from each read. Default: 1\r\n",
      "    -O MINLENGTH, --overlap=MINLENGTH\r\n",
      "                        Require MINLENGTH overlap between read and adapter for\r\n",
      "                        an adapter to be found. Default: 3\r\n",
      "    --match-read-wildcards\r\n",
      "                        Interpret IUPAC wildcards in reads. Default: False\r\n",
      "    -N, --no-match-adapter-wildcards\r\n",
      "                        Do not interpret IUPAC wildcards in adapters.\r\n",
      "    --no-trim           Match and redirect reads to output/untrimmed-output as\r\n",
      "                        usual, but do not remove adapters.\r\n",
      "    --mask-adapter      Mask adapters with 'N's instead of trimming them.\r\n",
      "\r\n",
      "  Additional read modifications:\r\n",
      "    -u LENGTH, --cut=LENGTH\r\n",
      "                        Remove bases from each read (first read only if\r\n",
      "                        paired). If LENGTH is positive, remove bases from the\r\n",
      "                        beginning. If LENGTH is negative, remove bases from\r\n",
      "                        the end. Can be used twice if LENGTHs have different\r\n",
      "                        signs. This is applied *before* adapter trimming.\r\n",
      "    --nextseq-trim=3'CUTOFF\r\n",
      "                        NextSeq-specific quality trimming (each read). Trims\r\n",
      "                        also dark cycles appearing as high-quality G bases.\r\n",
      "    -q [5'CUTOFF,]3'CUTOFF, --quality-cutoff=[5'CUTOFF,]3'CUTOFF\r\n",
      "                        Trim low-quality bases from 5' and/or 3' ends of each\r\n",
      "                        read before adapter removal. Applied to both reads if\r\n",
      "                        data is paired. If one value is given, only the 3' end\r\n",
      "                        is trimmed. If two comma-separated cutoffs are given,\r\n",
      "                        the 5' end is trimmed with the first cutoff, the 3'\r\n",
      "                        end with the second.\r\n",
      "    --quality-base=N    Assume that quality values in FASTQ are encoded as\r\n",
      "                        ascii(quality + N). This needs to be set to 64 for\r\n",
      "                        some old Illumina FASTQ files. Default: 33\r\n",
      "    -l LENGTH, --length=LENGTH\r\n",
      "                        Shorten reads to LENGTH. Positive values remove bases\r\n",
      "                        at the end while negative ones remove bases at the\r\n",
      "                        beginning. This and the following modifications are\r\n",
      "                        applied after adapter trimming.\r\n",
      "    --trim-n            Trim N's on ends of reads.\r\n",
      "    --length-tag=TAG    Search for TAG followed by a decimal number in the\r\n",
      "                        description field of the read. Replace the decimal\r\n",
      "                        number with the correct length of the trimmed read.\r\n",
      "                        For example, use --length-tag 'length=' to correct\r\n",
      "                        fields like 'length=123'.\r\n",
      "    --strip-suffix=STRIP_SUFFIX\r\n",
      "                        Remove this suffix from read names if present. Can be\r\n",
      "                        given multiple times.\r\n",
      "    -x PREFIX, --prefix=PREFIX\r\n",
      "                        Add this prefix to read names. Use {name} to insert\r\n",
      "                        the name of the matching adapter.\r\n",
      "    -y SUFFIX, --suffix=SUFFIX\r\n",
      "                        Add this suffix to read names; can also include {name}\r\n",
      "\r\n",
      "  Filtering of processed reads:\r\n",
      "    Filters are applied after above read modifications. Paired-end reads\r\n",
      "    are always discarded pairwise (see also --pair-filter).\r\n",
      "\r\n",
      "    -m LENGTH, --minimum-length=LENGTH\r\n",
      "                        Discard reads shorter than LENGTH. Default: 0\r\n",
      "    -M LENGTH, --maximum-length=LENGTH\r\n",
      "                        Discard reads longer than LENGTH. Default: no limit\r\n",
      "    --max-n=COUNT       Discard reads with more than COUNT 'N' bases. If COUNT\r\n",
      "                        is a number between 0 and 1, it is interpreted as a\r\n",
      "                        fraction of the read length.\r\n",
      "    --discard-trimmed, --discard\r\n",
      "                        Discard reads that contain an adapter. Also use -O to\r\n",
      "                        avoid discarding too many randomly matching reads!\r\n",
      "    --discard-untrimmed, --trimmed-only\r\n",
      "                        Discard reads that do not contain an adapter.\r\n",
      "    --discard-casava    Discard reads that did not pass CASAVA filtering\r\n",
      "                        (header has :Y:).\r\n",
      "\r\n",
      "  Output:\r\n",
      "    --quiet             Print only error messages.\r\n",
      "    -o FILE, --output=FILE\r\n",
      "                        Write trimmed reads to FILE. FASTQ or FASTA format is\r\n",
      "                        chosen depending on input. The summary report is sent\r\n",
      "                        to standard output. Use '{name}' in FILE to\r\n",
      "                        demultiplex reads into multiple files. Default: write\r\n",
      "                        to standard output\r\n",
      "    --info-file=FILE    Write information about each read and its adapter\r\n",
      "                        matches into FILE. See the documentation for the file\r\n",
      "                        format.\r\n",
      "    -r FILE, --rest-file=FILE\r\n",
      "                        When the adapter matches in the middle of a read,\r\n",
      "                        write the rest (after the adapter) to FILE.\r\n",
      "    --wildcard-file=FILE\r\n",
      "                        When the adapter has N wildcard bases, write adapter\r\n",
      "                        bases matching wildcard positions to FILE. (Inaccurate\r\n",
      "                        with indels.)\r\n",
      "    --too-short-output=FILE\r\n",
      "                        Write reads that are too short (according to length\r\n",
      "                        specified by -m) to FILE. Default: discard reads\r\n",
      "    --too-long-output=FILE\r\n",
      "                        Write reads that are too long (according to length\r\n",
      "                        specified by -M) to FILE. Default: discard reads\r\n",
      "    --untrimmed-output=FILE\r\n",
      "                        Write reads that do not contain any adapter to FILE.\r\n",
      "                        Default: output to same file as trimmed reads\r\n",
      "\r\n",
      "  Colorspace options:\r\n",
      "    -c, --colorspace    Enable colorspace mode\r\n",
      "    -d, --double-encode\r\n",
      "                        Double-encode colors (map 0,1,2,3,4 to A,C,G,T,N).\r\n",
      "    -t, --trim-primer   Trim primer base and the first color\r\n",
      "    --strip-f3          Strip the _F3 suffix of read names\r\n",
      "    --maq, --bwa        MAQ- and BWA-compatible colorspace output. This\r\n",
      "                        enables -c, -d, -t, --strip-f3 and -y '/1'.\r\n",
      "    -z, --zero-cap      Change negative quality values to zero. Enabled by\r\n",
      "                        default in colorspace mode since many tools have\r\n",
      "                        problems with negative qualities\r\n",
      "    --no-zero-cap       Disable zero capping\r\n",
      "\r\n",
      "  Paired-end options:\r\n",
      "    The -A/-G/-B/-U options work like their -a/-b/-g/-u counterparts, but\r\n",
      "    are applied to the second read in each pair.\r\n",
      "\r\n",
      "    -A ADAPTER          3' adapter to be removed from second read in a pair.\r\n",
      "    -G ADAPTER          5' adapter to be removed from second read in a pair.\r\n",
      "    -B ADAPTER          5'/3 adapter to be removed from second read in a pair.\r\n",
      "    -U LENGTH           Remove LENGTH bases from second read in a pair.\r\n",
      "    -p FILE, --paired-output=FILE\r\n",
      "                        Write second read in a pair to FILE.\r\n",
      "    --pair-filter=(any|both)\r\n",
      "                        Which of the reads in a paired-end read have to match\r\n",
      "                        the filtering criterion in order for the pair to be\r\n",
      "                        filtered. Default: any\r\n",
      "    --interleaved       Read and write interleaved paired-end reads.\r\n",
      "    --untrimmed-paired-output=FILE\r\n",
      "                        Write second read in a pair to this FILE when no\r\n",
      "                        adapter was found. Use with --untrimmed-output.\r\n",
      "                        Default: output to same file as trimmed reads\r\n",
      "    --too-short-paired-output=FILE\r\n",
      "                        Write second read in a pair to this file if pair is\r\n",
      "                        too short. Use also --too-short-output.\r\n",
      "    --too-long-paired-output=FILE\r\n",
      "                        Write second read in a pair to this file if pair is\r\n",
      "                        too long. Use also --too-long-output.\r\n"
     ]
    }
   ],
   "source": [
    "!cutadapt --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cutadapt version 1.17\r\n",
      "Copyright (C) 2010-2018 Marcel Martin <marcel.martin@scilifelab.se>\r\n",
      "\r\n",
      "cutadapt removes adapter sequences from high-throughput sequencing reads.\r\n",
      "\r\n",
      "Usage:\r\n",
      "    cutadapt -a ADAPTER [options] [-o output.fastq] input.fastq\r\n",
      "\r\n",
      "For paired-end reads:\r\n",
      "    cutadapt -a ADAPT1 -A ADAPT2 [options] -o out1.fastq -p out2.fastq in1.fastq in2.fastq\r\n",
      "\r\n",
      "Replace \"ADAPTER\" with the actual sequence of your 3' adapter. IUPAC wildcard\r\n",
      "characters are supported. The reverse complement is *not* automatically\r\n",
      "searched. All reads from input.fastq will be written to output.fastq with the\r\n",
      "adapter sequence removed. Adapter matching is error-tolerant. Multiple adapter\r\n",
      "sequences can be given (use further -a options), but only the best-matching\r\n",
      "adapter will be removed.\r\n",
      "\r\n",
      "Input may also be in FASTA format. Compressed input and output is supported and\r\n",
      "auto-detected from the file name (.gz, .xz, .bz2). Use the file name '-' for\r\n",
      "standard input/output. Without the -o option, output is sent to standard output.\r\n",
      "\r\n",
      "Citation:\r\n",
      "\r\n",
      "Marcel Martin. Cutadapt removes adapter sequences from high-throughput\r\n",
      "sequencing reads. EMBnet.Journal, 17(1):10-12, May 2011.\r\n",
      "http://dx.doi.org/10.14806/ej.17.1.200\r\n",
      "\r\n",
      "Use \"cutadapt --help\" to see all command-line options.\r\n",
      "See http://cutadapt.readthedocs.io/ for full documentation.\r\n",
      "\r\n",
      "cutadapt: error: Character '/' in adapter sequence '/TSERS/KELLYCRIBARI/DOCTMENTS/BIOINFORMATICS/GITHTB/CRIBARI-EDNA/FASTA/METADATATAGS.FASTA' is not a valid IUPAC code. Use only characters XACGTURYSWKMBDHVN.\r\n"
     ]
    }
   ],
   "source": [
    "!cutadapt \\\n",
    "file: /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/FASTA/Lib-1.assembled.fasta \\\n",
    "-g /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/FASTA/metadatatags.fasta \\\n",
    "-o /Users/kellycribari/Documents/Bioinformatics/GitHub/Cribari-eDNA/Demultiplexed/sample-{name}.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
