{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun\r\n"
     ]
    }
   ],
   "source": [
    "!git clone https://github.com/jimmyodonnell/banzai.git"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use the git desktop to clone the pipeline into /users/kellycri/banzai-master\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run the test dataset of the pipeline to check everything is working ok\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\t  üèÑ   ¬° BANZAI !\n",
      "\n",
      "\n",
      "Reading analysis parameters from:\n",
      "/Users/kellycribari/banzai-master/banzai_params.sh\n",
      "\n",
      "Reading sequencing metadata from:\n",
      "/Users/kellycribari/banzai-master/data/PS_16S/metadata.csv\n",
      "\n",
      "The file passes test for CRLF. Everybody dance!\n",
      "\n",
      "Checking for dependencies: pear cutadapt vsearch swarm seqtk python blastn R\n",
      "Found program pear in /usr/local/bin/pear\n",
      "Found program cutadapt in /usr/local/bin/cutadapt\n",
      "Found program vsearch in /usr/local/bin/vsearch\n",
      "Found program swarm in /usr/local/bin/swarm\n",
      "Found program seqtk in /usr/local/bin/seqtk\n",
      "Found program python in /Users/kellycribari/anaconda3/bin/python\n",
      "Found program blastn in /usr/local/ncbi/blast/bin/blastn\n",
      "Found program R in /usr/local/bin/R\n",
      "pigz installation not found; using gzip\n",
      "\n",
      "4 cores detected.\n",
      "\n",
      "2018-11-02 11:06 Analysis started at 20181102_1106\n",
      "Output is located in:\n",
      "/Users/kellycribari/banzai_out_20181102_1106\n",
      "\n",
      "Metadata has 280 rows and 16 columns including header.\n",
      "Expecting 279 samples total.\n",
      "\n",
      "Files read from metadata columns 15 and 16\n",
      "File names:\n",
      "A-sp-GTGAAA_S5_R1_001_sub.fastq\tA-sp-GTGAAA_S5_R2_001_sub.fastq\n",
      "B-sp-sp-TAGCTT_S6_R1_001_sub.fastq\tB-sp-sp-TAGCTT_S6_R2_001_sub.fastq\n",
      "C-sp-sp-TCCCGA_S7_R1_001_sub.fastq\tC-sp-sp-TCCCGA_S7_R2_001_sub.fastq\n",
      "D-ATTCCT_S8_R1_001_sub.fastq\tD-ATTCCT_S8_R2_001_sub.fastq\n",
      "E-sp-CCGTCC_S9_R1_001_sub.fastq\tE-sp-CCGTCC_S9_R2_001_sub.fastq\n",
      "F-CTATAC_S10_R1_001_sub.fastq\tF-CTATAC_S10_R2_001_sub.fastq\n",
      "G-sp-sp-AGTCAA_S11_R1_001_sub.fastq\tG-sp-sp-AGTCAA_S11_R2_001_sub.fastq\n",
      "H-GTTTCG_S12_R1_001_sub.fastq\tH-GTTTCG_S12_R2_001_sub.fastq\n",
      "J-sp-CAACTA_S13_R1_001_sub.fastq\tJ-sp-CAACTA_S13_R2_001_sub.fastq\n",
      "K-sp-sp-GGTAGC_S14_R1_001_sub.fastq\tK-sp-sp-GGTAGC_S14_R2_001_sub.fastq\n",
      "\n",
      "Secondary indexes read from sequencing metadata (6 total)\n",
      "\n",
      "Primers read from metadata columns 13 and 14\n",
      "Primer sequences: AGTTACYYTAGGGATAACAGCG CCGGTCTGAACTCAGATCAYGT\n",
      "\n",
      "2018-11-02 11:06 Merging reads in library A...\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/A/A-sp-GTGAAA_S5_R1_001_sub.fastq\n",
      "Reverse reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/A/A-sp-GTGAAA_S5_R2_001_sub.fastq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 10000\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 15\n",
      "Minimum read size after trimming...: 75\n",
      "Maximal ratio of uncalled bases....: 0.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 4\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.275563\n",
      "  C: 0.223057\n",
      "  G: 0.229077\n",
      "  T: 0.272304\n",
      "  257 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 932 / 1,000 (93.200%)\n",
      "Discarded reads ...................: 68 / 1,000 (6.800%)\n",
      "Not assembled reads ...............: 0 / 1,000 (0.000%)\n",
      "Assembled reads file...............: /Users/kellycribari/banzai_out_20181102_1106/A/1_merged.assembled.fastq\n",
      "Discarded reads file...............: /Users/kellycribari/banzai_out_20181102_1106/A/1_merged.discarded.fastq\n",
      "Unassembled forward reads file.....: /Users/kellycribari/banzai_out_20181102_1106/A/1_merged.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /Users/kellycribari/banzai_out_20181102_1106/A/1_merged.unassembled.reverse.fastq\n",
      "\n",
      "2018-11-02 11:06 Filtering merged reads...\n",
      "vsearch v2.8.2_macos_x86_64, 8.0GB RAM, 4 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%\n",
      "908 sequences kept (of which 0 truncated), 24 sequences discarded.\n",
      "\n",
      "2018-11-02 11:06 Compressing PEAR output...\n",
      "2018-11-02 11:06 PEAR output compressed.\n",
      "\n",
      "Reads not renamed\n",
      "\n",
      "Homopolymers not removed.\n",
      "\n",
      "2018-11-02 11:06 Concatenating fasta files...\n",
      "\n",
      "2018-11-02 11:06 Merging reads in library B...\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/B/B-sp-sp-TAGCTT_S6_R1_001_sub.fastq\n",
      "Reverse reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/B/B-sp-sp-TAGCTT_S6_R2_001_sub.fastq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 10000\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 15\n",
      "Minimum read size after trimming...: 75\n",
      "Maximal ratio of uncalled bases....: 0.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 4\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.278081\n",
      "  C: 0.220253\n",
      "  G: 0.225461\n",
      "  T: 0.276205\n",
      "  245 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 930 / 1,000 (93.000%)\n",
      "Discarded reads ...................: 70 / 1,000 (7.000%)\n",
      "Not assembled reads ...............: 0 / 1,000 (0.000%)\n",
      "Assembled reads file...............: /Users/kellycribari/banzai_out_20181102_1106/B/1_merged.assembled.fastq\n",
      "Discarded reads file...............: /Users/kellycribari/banzai_out_20181102_1106/B/1_merged.discarded.fastq\n",
      "Unassembled forward reads file.....: /Users/kellycribari/banzai_out_20181102_1106/B/1_merged.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /Users/kellycribari/banzai_out_20181102_1106/B/1_merged.unassembled.reverse.fastq\n",
      "\n",
      "2018-11-02 11:06 Filtering merged reads...\n",
      "vsearch v2.8.2_macos_x86_64, 8.0GB RAM, 4 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%\n",
      "911 sequences kept (of which 0 truncated), 19 sequences discarded.\n",
      "\n",
      "2018-11-02 11:06 Compressing PEAR output...\n",
      "2018-11-02 11:06 PEAR output compressed.\n",
      "\n",
      "Reads not renamed\n",
      "\n",
      "Homopolymers not removed.\n",
      "\n",
      "2018-11-02 11:06 Concatenating fasta files...\n",
      "\n",
      "2018-11-02 11:06 Merging reads in library C...\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/C/C-sp-sp-TCCCGA_S7_R1_001_sub.fastq\n",
      "Reverse reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/C/C-sp-sp-TCCCGA_S7_R2_001_sub.fastq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 10000\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 15\n",
      "Minimum read size after trimming...: 75\n",
      "Maximal ratio of uncalled bases....: 0.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 4\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.278908\n",
      "  C: 0.218999\n",
      "  G: 0.224969\n",
      "  T: 0.277124\n",
      "  322 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 905 / 1,000 (90.500%)\n",
      "Discarded reads ...................: 95 / 1,000 (9.500%)\n",
      "Not assembled reads ...............: 0 / 1,000 (0.000%)\n",
      "Assembled reads file...............: /Users/kellycribari/banzai_out_20181102_1106/C/1_merged.assembled.fastq\n",
      "Discarded reads file...............: /Users/kellycribari/banzai_out_20181102_1106/C/1_merged.discarded.fastq\n",
      "Unassembled forward reads file.....: /Users/kellycribari/banzai_out_20181102_1106/C/1_merged.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /Users/kellycribari/banzai_out_20181102_1106/C/1_merged.unassembled.reverse.fastq\n",
      "\n",
      "2018-11-02 11:06 Filtering merged reads...\n",
      "vsearch v2.8.2_macos_x86_64, 8.0GB RAM, 4 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%\n",
      "883 sequences kept (of which 0 truncated), 22 sequences discarded.\n",
      "\n",
      "2018-11-02 11:06 Compressing PEAR output...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2018-11-02 11:06 PEAR output compressed.\n",
      "\n",
      "Reads not renamed\n",
      "\n",
      "Homopolymers not removed.\n",
      "\n",
      "2018-11-02 11:06 Concatenating fasta files...\n",
      "\n",
      "2018-11-02 11:06 Merging reads in library D...\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/D/D-ATTCCT_S8_R1_001_sub.fastq\n",
      "Reverse reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/D/D-ATTCCT_S8_R2_001_sub.fastq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 10000\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 15\n",
      "Minimum read size after trimming...: 75\n",
      "Maximal ratio of uncalled bases....: 0.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 4\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.280494\n",
      "  C: 0.218231\n",
      "  G: 0.223716\n",
      "  T: 0.277559\n",
      "  164 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 950 / 1,000 (95.000%)\n",
      "Discarded reads ...................: 50 / 1,000 (5.000%)\n",
      "Not assembled reads ...............: 0 / 1,000 (0.000%)\n",
      "Assembled reads file...............: /Users/kellycribari/banzai_out_20181102_1106/D/1_merged.assembled.fastq\n",
      "Discarded reads file...............: /Users/kellycribari/banzai_out_20181102_1106/D/1_merged.discarded.fastq\n",
      "Unassembled forward reads file.....: /Users/kellycribari/banzai_out_20181102_1106/D/1_merged.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /Users/kellycribari/banzai_out_20181102_1106/D/1_merged.unassembled.reverse.fastq\n",
      "\n",
      "2018-11-02 11:06 Filtering merged reads...\n",
      "vsearch v2.8.2_macos_x86_64, 8.0GB RAM, 4 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%\n",
      "930 sequences kept (of which 0 truncated), 20 sequences discarded.\n",
      "\n",
      "2018-11-02 11:06 Compressing PEAR output...\n",
      "2018-11-02 11:06 PEAR output compressed.\n",
      "\n",
      "Reads not renamed\n",
      "\n",
      "Homopolymers not removed.\n",
      "\n",
      "2018-11-02 11:06 Concatenating fasta files...\n",
      "\n",
      "2018-11-02 11:06 Merging reads in library E...\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/E/E-sp-CCGTCC_S9_R1_001_sub.fastq\n",
      "Reverse reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/E/E-sp-CCGTCC_S9_R2_001_sub.fastq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 10000\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 15\n",
      "Minimum read size after trimming...: 75\n",
      "Maximal ratio of uncalled bases....: 0.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 4\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.273668\n",
      "  C: 0.225404\n",
      "  G: 0.230893\n",
      "  T: 0.270035\n",
      "  243 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 935 / 1,000 (93.500%)\n",
      "Discarded reads ...................: 65 / 1,000 (6.500%)\n",
      "Not assembled reads ...............: 0 / 1,000 (0.000%)\n",
      "Assembled reads file...............: /Users/kellycribari/banzai_out_20181102_1106/E/1_merged.assembled.fastq\n",
      "Discarded reads file...............: /Users/kellycribari/banzai_out_20181102_1106/E/1_merged.discarded.fastq\n",
      "Unassembled forward reads file.....: /Users/kellycribari/banzai_out_20181102_1106/E/1_merged.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /Users/kellycribari/banzai_out_20181102_1106/E/1_merged.unassembled.reverse.fastq\n",
      "\n",
      "2018-11-02 11:06 Filtering merged reads...\n",
      "vsearch v2.8.2_macos_x86_64, 8.0GB RAM, 4 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%\n",
      "912 sequences kept (of which 0 truncated), 23 sequences discarded.\n",
      "\n",
      "2018-11-02 11:06 Compressing PEAR output...\n",
      "2018-11-02 11:06 PEAR output compressed.\n",
      "\n",
      "Reads not renamed\n",
      "\n",
      "Homopolymers not removed.\n",
      "\n",
      "2018-11-02 11:06 Concatenating fasta files...\n",
      "\n",
      "2018-11-02 11:06 Merging reads in library F...\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/F/F-CTATAC_S10_R1_001_sub.fastq\n",
      "Reverse reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/F/F-CTATAC_S10_R2_001_sub.fastq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 10000\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 15\n",
      "Minimum read size after trimming...: 75\n",
      "Maximal ratio of uncalled bases....: 0.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 4\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.280140\n",
      "  C: 0.217883\n",
      "  G: 0.224351\n",
      "  T: 0.277626\n",
      "  309 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 912 / 1,000 (91.200%)\n",
      "Discarded reads ...................: 88 / 1,000 (8.800%)\n",
      "Not assembled reads ...............: 0 / 1,000 (0.000%)\n",
      "Assembled reads file...............: /Users/kellycribari/banzai_out_20181102_1106/F/1_merged.assembled.fastq\n",
      "Discarded reads file...............: /Users/kellycribari/banzai_out_20181102_1106/F/1_merged.discarded.fastq\n",
      "Unassembled forward reads file.....: /Users/kellycribari/banzai_out_20181102_1106/F/1_merged.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /Users/kellycribari/banzai_out_20181102_1106/F/1_merged.unassembled.reverse.fastq\n",
      "\n",
      "2018-11-02 11:06 Filtering merged reads...\n",
      "vsearch v2.8.2_macos_x86_64, 8.0GB RAM, 4 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%\n",
      "882 sequences kept (of which 0 truncated), 30 sequences discarded.\n",
      "\n",
      "2018-11-02 11:06 Compressing PEAR output...\n",
      "2018-11-02 11:06 PEAR output compressed.\n",
      "\n",
      "Reads not renamed\n",
      "\n",
      "Homopolymers not removed.\n",
      "\n",
      "2018-11-02 11:06 Concatenating fasta files...\n",
      "\n",
      "2018-11-02 11:06 Merging reads in library G...\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/G/G-sp-sp-AGTCAA_S11_R1_001_sub.fastq\n",
      "Reverse reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/G/G-sp-sp-AGTCAA_S11_R2_001_sub.fastq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 10000\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 15\n",
      "Minimum read size after trimming...: 75\n",
      "Maximal ratio of uncalled bases....: 0.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 4\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.280638\n",
      "  C: 0.217187\n",
      "  G: 0.223517\n",
      "  T: 0.278659\n",
      "  127 uncalled bases\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 962 / 1,000 (96.200%)\n",
      "Discarded reads ...................: 38 / 1,000 (3.800%)\n",
      "Not assembled reads ...............: 0 / 1,000 (0.000%)\n",
      "Assembled reads file...............: /Users/kellycribari/banzai_out_20181102_1106/G/1_merged.assembled.fastq\n",
      "Discarded reads file...............: /Users/kellycribari/banzai_out_20181102_1106/G/1_merged.discarded.fastq\n",
      "Unassembled forward reads file.....: /Users/kellycribari/banzai_out_20181102_1106/G/1_merged.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /Users/kellycribari/banzai_out_20181102_1106/G/1_merged.unassembled.reverse.fastq\n",
      "\n",
      "2018-11-02 11:06 Filtering merged reads...\n",
      "vsearch v2.8.2_macos_x86_64, 8.0GB RAM, 4 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%\n",
      "944 sequences kept (of which 0 truncated), 18 sequences discarded.\n",
      "\n",
      "2018-11-02 11:06 Compressing PEAR output...\n",
      "2018-11-02 11:06 PEAR output compressed.\n",
      "\n",
      "Reads not renamed\n",
      "\n",
      "Homopolymers not removed.\n",
      "\n",
      "2018-11-02 11:06 Concatenating fasta files...\n",
      "\n",
      "2018-11-02 11:06 Merging reads in library H...\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/H/H-GTTTCG_S12_R1_001_sub.fastq\n",
      "Reverse reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/H/H-GTTTCG_S12_R2_001_sub.fastq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 10000\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 15\n",
      "Minimum read size after trimming...: 75\n",
      "Maximal ratio of uncalled bases....: 0.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 4\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.280721\n",
      "  C: 0.217263\n",
      "  G: 0.223063\n",
      "  T: 0.278953\n",
      "  251 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 932 / 1,000 (93.200%)\n",
      "Discarded reads ...................: 68 / 1,000 (6.800%)\n",
      "Not assembled reads ...............: 0 / 1,000 (0.000%)\n",
      "Assembled reads file...............: /Users/kellycribari/banzai_out_20181102_1106/H/1_merged.assembled.fastq\n",
      "Discarded reads file...............: /Users/kellycribari/banzai_out_20181102_1106/H/1_merged.discarded.fastq\n",
      "Unassembled forward reads file.....: /Users/kellycribari/banzai_out_20181102_1106/H/1_merged.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /Users/kellycribari/banzai_out_20181102_1106/H/1_merged.unassembled.reverse.fastq\n",
      "\n",
      "2018-11-02 11:06 Filtering merged reads...\n",
      "vsearch v2.8.2_macos_x86_64, 8.0GB RAM, 4 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%\n",
      "908 sequences kept (of which 0 truncated), 24 sequences discarded.\n",
      "\n",
      "2018-11-02 11:06 Compressing PEAR output...\n",
      "2018-11-02 11:06 PEAR output compressed.\n",
      "\n",
      "Reads not renamed\n",
      "\n",
      "Homopolymers not removed.\n",
      "\n",
      "2018-11-02 11:06 Concatenating fasta files...\n",
      "\n",
      "2018-11-02 11:06 Merging reads in library J...\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/J/J-sp-CAACTA_S13_R1_001_sub.fastq\n",
      "Reverse reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/J/J-sp-CAACTA_S13_R2_001_sub.fastq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 10000\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 15\n",
      "Minimum read size after trimming...: 75\n",
      "Maximal ratio of uncalled bases....: 0.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 4\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.276054\n",
      "  C: 0.222605\n",
      "  G: 0.228968\n",
      "  T: 0.272373\n",
      "  305 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 923 / 1,000 (92.300%)\n",
      "Discarded reads ...................: 77 / 1,000 (7.700%)\n",
      "Not assembled reads ...............: 0 / 1,000 (0.000%)\n",
      "Assembled reads file...............: /Users/kellycribari/banzai_out_20181102_1106/J/1_merged.assembled.fastq\n",
      "Discarded reads file...............: /Users/kellycribari/banzai_out_20181102_1106/J/1_merged.discarded.fastq\n",
      "Unassembled forward reads file.....: /Users/kellycribari/banzai_out_20181102_1106/J/1_merged.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /Users/kellycribari/banzai_out_20181102_1106/J/1_merged.unassembled.reverse.fastq\n",
      "\n",
      "2018-11-02 11:06 Filtering merged reads...\n",
      "vsearch v2.8.2_macos_x86_64, 8.0GB RAM, 4 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%\n",
      "900 sequences kept (of which 0 truncated), 23 sequences discarded.\n",
      "\n",
      "2018-11-02 11:06 Compressing PEAR output...\n",
      "2018-11-02 11:06 PEAR output compressed.\n",
      "\n",
      "Reads not renamed\n",
      "\n",
      "Homopolymers not removed.\n",
      "\n",
      "2018-11-02 11:06 Concatenating fasta files...\n",
      "\n",
      "2018-11-02 11:06 Merging reads in library K...\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/K/K-sp-sp-GGTAGC_S14_R1_001_sub.fastq\n",
      "Reverse reads file.................: /Users/kellycribari/banzai-master/data/PS_16S/K/K-sp-sp-GGTAGC_S14_R2_001_sub.fastq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 10000\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 15\n",
      "Minimum read size after trimming...: 75\n",
      "Maximal ratio of uncalled bases....: 0.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 4\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.277428\n",
      "  C: 0.220408\n",
      "  G: 0.226159\n",
      "  T: 0.276005\n",
      "  278 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 919 / 1,000 (91.900%)\n",
      "Discarded reads ...................: 81 / 1,000 (8.100%)\n",
      "Not assembled reads ...............: 0 / 1,000 (0.000%)\n",
      "Assembled reads file...............: /Users/kellycribari/banzai_out_20181102_1106/K/1_merged.assembled.fastq\n",
      "Discarded reads file...............: /Users/kellycribari/banzai_out_20181102_1106/K/1_merged.discarded.fastq\n",
      "Unassembled forward reads file.....: /Users/kellycribari/banzai_out_20181102_1106/K/1_merged.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /Users/kellycribari/banzai_out_20181102_1106/K/1_merged.unassembled.reverse.fastq\n",
      "\n",
      "2018-11-02 11:06 Filtering merged reads...\n",
      "vsearch v2.8.2_macos_x86_64, 8.0GB RAM, 4 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%\n",
      "889 sequences kept (of which 0 truncated), 30 sequences discarded.\n",
      "\n",
      "2018-11-02 11:06 Compressing PEAR output...\n",
      "2018-11-02 11:06 PEAR output compressed.\n",
      "\n",
      "Reads not renamed\n",
      "\n",
      "Homopolymers not removed.\n",
      "\n",
      "2018-11-02 11:06 Concatenating fasta files...\n",
      "\n",
      "2018-11-02 11:06 Counting sequences in primer removal input...\n",
      "2018-11-02 11:06 9067 sequences found in primer removal input\n",
      "\n",
      "2018-11-02 11:06 Beginning primer removal...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 1.17 with Python 2.7.10\n",
      "This is cutadapt 1.17 with Python 2.7.10\n",
      "Command line parameters: -g ^CCGGTCTGAACTCAGATCAYGT -e 0.10 -m 62 --discard-untrimmed /Users/kellycribari/banzai_out_20181102_1106/all_lib/demult_concat.fasta\n",
      "Command line parameters: -g ^AGTTACYYTAGGGATAACAGCG -e 0.10 -m 62 --discard-untrimmed /Users/kellycribari/banzai_out_20181102_1106/all_lib/demult_concat.fasta\n",
      "Running on 1 core\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.25 s (28 us/read; 2.15 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   9,067\n",
      "Reads with adapters:                     4,049 (44.7%)\n",
      "Reads that were too short:                   4 (0.0%)\n",
      "Reads written (passing filters):         4,047 (44.6%)\n",
      "\n",
      "Total basepairs processed:     1,415,231 bp\n",
      "Total written (filtered):        546,395 bp (38.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CCGGTCTGAACTCAGATCAYGT; Type: anchored 5'; Length: 22; Trimmed: 4049 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t30\t0.0\t2\t0 0 30\n",
      "21\t83\t0.0\t2\t0 82 1\n",
      "22\t3922\t0.0\t2\t3860 59 3\n",
      "23\t14\t0.0\t2\t0 14\n",
      "\n",
      "Finished in 0.26 s (29 us/read; 2.09 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   9,067\n",
      "Reads with adapters:                     4,395 (48.5%)\n",
      "Reads that were too short:                   5 (0.1%)\n",
      "Reads written (passing filters):         4,393 (48.5%)\n",
      "\n",
      "Total basepairs processed:     1,415,231 bp\n",
      "Total written (filtered):        593,135 bp (41.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGTTACYYTAGGGATAACAGCG; Type: anchored 5'; Length: 22; Trimmed: 4395 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t60\t0.0\t2\t0 0 60\n",
      "21\t96\t0.0\t2\t0 94 2\n",
      "22\t4227\t0.0\t2\t4191 35 1\n",
      "23\t12\t0.0\t2\t0 12\n",
      "\n",
      "This is cutadapt 1.17 with Python 2.7.10\n",
      "Command line parameters: -a ACRTGATCTGAGTTCAGACCGG$ -e 0.10 -m 62 --discard-untrimmed /Users/kellycribari/banzai_out_20181102_1106/all_lib/5_primerL1_removed.fasta\n",
      "This is cutadapt 1.17 with Python 2.7.10\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Command line parameters: -a CGCTGTTATCCCTARRGTAACT$ -e 0.10 -m 62 --discard-untrimmed /Users/kellycribari/banzai_out_20181102_1106/all_lib/5_primerL2_removed.fasta\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.17 s (41 us/read; 1.46 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   4,047\n",
      "Reads with adapters:                     3,773 (93.2%)\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):         3,773 (93.2%)\n",
      "\n",
      "Total basepairs processed:       546,395 bp\n",
      "Total written (filtered):        430,003 bp (78.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CGCTGTTATCCCTARRGTAACT; Type: anchored 3'; Length: 22; Trimmed: 3773 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 44.4%\n",
      "  C: 0.0%\n",
      "  G: 55.5%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t40\t0.0\t2\t0 0 40\n",
      "21\t91\t0.0\t2\t0 89 2\n",
      "22\t3626\t0.0\t2\t3585 41\n",
      "23\t16\t0.0\t2\t0 16\n",
      "\n",
      "Finished in 0.18 s (41 us/read; 1.45 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                   4,393\n",
      "Reads with adapters:                     4,105 (93.4%)\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):         4,105 (93.4%)\n",
      "\n",
      "Total basepairs processed:       593,135 bp\n",
      "Total written (filtered):        466,774 bp (78.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ACRTGATCTGAGTTCAGACCGG; Type: anchored 3'; Length: 22; Trimmed: 4105 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 32.3%\n",
      "  C: 2.1%\n",
      "  G: 0.2%\n",
      "  T: 65.5%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t25\t0.0\t2\t0 0 25\n",
      "21\t92\t0.0\t2\t0 91 1\n",
      "22\t3978\t0.0\t2\t3913 63 2\n",
      "23\t8\t0.0\t2\t0 8\n",
      "24\t2\t0.0\t2\t0 0 2\n",
      "\n",
      "2018-11-02 11:06 Correcting sequence orientation...\n",
      "2018-11-02 11:06 Compressing primer removal files...\n",
      "2018-11-02 11:06 Primer removal files compressed.\n",
      "\n",
      "2018-11-02 11:06 Identifying identical sequences... (python)\n",
      "  File \"/Users/kellycribari/banzai-master/scripts/dereplication/derep_fasta.py\", line 120\n",
      "    f.write('\\n'.join([\n",
      "                      ^\n",
      "TabError: inconsistent use of tabs and spaces in indentation\n",
      "There was a problem generating the duplicate fasta file. It is empty or absent.\n",
      "The remainder of the script depends on this file.\n",
      "Aborting script.\n"
     ]
    }
   ],
   "source": [
    "!bash /Users/kellycribari/banzai-master/banzai.sh test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
